### PR TITLE
Created a JavaScript function to return a JSON object and displayed the data in HTML

### PR DESCRIPTION
**Pull request by gracieorozco**
I created another JavaScript function that utilizes fetch() to retrieve an ArrayList<String> encoded in a JSON object. I then displayed the data with HTML using a JavaScript function. This is the third pull request out of five for this week.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,9 +26,32 @@
...
 public class DataServlet extends HttpServlet {
 
+    public ArrayList<String> messages;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // below command was used in servlet and fetch() tutorials
+    // response.setContentType("text/html;");
```
Comment (KaylaNguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Remove unused code.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,9 +26,32 @@
...
+    messages.add("Hello");
+    messages.add("Hi");
+    messages.add("Greetings");
+
+    String json = convertToJsonUsingGson(messages);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public String convertToJsonUsingGson(ArrayList<String> m) {
```
Comment (KaylaNguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Let's name this variable something more descriptive.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -28,3 +28,46 @@ function addRandomFact() {
...
+ }
+
+/**
+ * Uses fetch() to obtain json data from DataServlet.java
+ */
+ function obtain_json_data() {
+    fetch("/data")
+        .then(response => response.json())
+        .then(messages => {
+         console.log(messages)
```
Comment (KaylaNguyen) [inline portfolio/src/main/webapp/script.js]: Format: add space.

Comment (KaylaNguyen) [reply portfolio/src/main/webapp/script.js]: Also remove debugging logs.

**Comment by gracieorozco** [comment]
I've updated my website with the changes made for pull request 3. 
http://gorozco-sps-spring20.appspot.com/

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -24,9 +26,35 @@
...
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+    comment_list.add(text);
+
+    // Respond with the result.
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("New Comment: " + text);
+    for (int i = 1; i < comment_list.size(); i++) {
+        response.getWriter().println("Comment: " + comment_list.get(i));
```
Comment (KaylaNguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You might want to add a new line after each comment for better display. Same for "New Comment" section above.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -226,6 +226,24 @@ <h3>2/18/20</h3>
       <p>Click here to get a random fact about myself:</p>
       <button onclick="addRandomFact()">Press</button>
       <div id="fact-container"></div>
+      <h2>Week 2</h2>
+      <!-- <p>Click <a href="/data">here</a> to try out a week 2 servlet page</p>
```
Comment (KaylaNguyen) [inline portfolio/src/main/webapp/index.html]: Remove unused code.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -24,9 +26,34 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    public ArrayList<String> comment_list = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    comment_list = new ArrayList<String>();
```
Comment (KaylaNguyen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You're reinitializing this comment_list here while on line 29, you already did.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (KaylaNguyen) [reply portfolio/src/main/webapp/script.js]: Can you remove console.log(messages)? It's used for debugging and shouldn't be submitted as part of the code.

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -28,3 +28,46 @@ function addRandomFact() {
...
+        })
+        .catch((test) => {
```
Comment (KaylaNguyen) [inline portfolio/src/main/webapp/script.js]: Remove redundant parenthesis.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (KaylaNguyen) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Reopen since the comment is not resolved.

**Pull request merged**